src/test/java/net/sf/jabref/logic/importer/fetcher/AstrophysicsDataSystemTest.java

```diff
@@ -75,4 +160,28 @@ public void searchByEntryFindsEntry() throws Exception {
         assertFalse(fetchedEntries.isEmpty());
         assertEquals(diezSliceTheoremEntry, fetchedEntries.get(0));
     }
+
+    @Test
+    public void testPerformSearchByFamaeyMcGaughEntry() throws Exception {
+        Optional<BibEntry> fetchedEntry = fetcher.performSearchById("10.12942/lrr-2012-10");
+        assertEquals(Optional.of(famaeyMcGaughEntry), fetchedEntry);
+    }
+
+    @Test
+    public void testPerformSearchByIdEmptyDOI() throws Exception {
+        Optional<BibEntry> fetchedEntry = fetcher.performSearchById("");
+        assertEquals(Optional.empty(), fetchedEntry);
+    }
+
+    @Test(expected = FetcherException.class)
+    public void testPerformSearchByIdInvalidDoi() throws Exception {
+        Optional<BibEntry> fetchedEntry = fetcher.performSearchById("this.doi.will.fail");
+        assertEquals(Optional.empty(), fetchedEntry);
```

Comment:
I think you then have to use the fail(), otherwise it will produce a codecov or warning in Eclipse for about the test